Project/Assets/_Scripts/_new_Scripts/Player Code/CharacterModel.cs

```diff
@@ -1,128 +1,63 @@
-		[XmlElement(ElementName = "Data")]
-		public CharacterData Data;
-
-		[XmlArray(ElementName = "MotionModes")]
-		public HashSet<MotionMode> MotionModes;
-		#endregion
-
-		#region Interface
-		/// <summary>
-		/// Calls all available movement modes.
-		/// </summary>
-		public void DoMotion()
-		{
-			MotionModes.Select((MotionMode mode) => mode.Do(ViewModel));
-		}
-		#endregion
-
-		#region Serialization
-		/// <summary>
-		/// TODO: I think that this method can be generalized to any type
-		/// </summary>
-		/// <param name="xmlPath"></param>
-		/// <param name="characterModel"></param>
-		public static void SaveToXml(string xmlPath, CharacterModel characterModel)
-		{
-			// Get all types for the serializer
-			var MotionModeTypes = characterModel.MotionModes.Select((MotionMode mode) => mode.GetType()).ToArray();
-
-			// Create a new serializer
-			var serializer = new XmlSerializer(characterModel.GetType(), extraTypes: MotionModeTypes);
-
-			// Open the file
-			using (var filestream = new FileStream(path: xmlPath,
-				  mode: FileMode.CreateNew))
-			{
-				serializer.Serialize(stream: filestream, o: characterModel);
-			}
-		}
-		/// <summary>
-		/// TODO: I think that this method can be generalized to any type
-		/// </summary>
-		/// <param name="xmlPath"></param>
-		/// <param name="extraSuperClasses"></param>
-		/// <returns></returns>
-		public static CharacterModel ReadFromXml(string xmlPath, Type[] extraSuperClasses)
-		{
-			var exportedTypes = Assembly.GetExecutingAssembly()
-				.GetExportedTypes();
-
-			// Get all types subclassing the types in the parameter
-			var extraTypes = extraSuperClasses.Select((Type superClass) =>
-			{
-				return exportedTypes
-				.Where((Type type) => type.IsSubclassOf(superClass))
-				.ToArray();
-			})
-			.SelectMany(x => x)
-			.Distinct()
-			.ToArray();
-
-			var serializer = new XmlSerializer(typeof(CharacterModel), extraTypes: extraTypes);
-
-			// Handle the unknown nodes and attributes
-			serializer.UnknownNode += new XmlNodeEventHandler(serializer.UnknownNodeCallback);
-			serializer.UnknownAttribute += new XmlAttributeEventHandler(serializer.UnknownAttributeCallback);
-
-			// Open the file
-			using (var filestream = new FileStream(path: xmlPath,
-				  mode: FileMode.Open))
-			{
-				// TODO: copy constructor
-				var deserializationResult = serializer.Deserialize(stream: filestream) as CharacterModel;
-				return deserializationResult;
-			}
-		}
-		#endregion
-	}
-}
-
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using Steamy.Player.MotionModes;
+using UnityEngine;
+
+namespace Steamy.Player
+{
+    public delegate void Callback();
+
+    [Serializable]
+    public class CharacterModel
+    {
+        public string Name;
+        public CharacterHealth Health;
+        public List<MotionMode> MotionModes;
+
+        public CharacterModel() { }
+    }
+
+    [Serializable]
+    public class CharacterHealth : INotifyPropertyChanged

```

Comment:
Mam nadzieję, że domyślnie jedna klasa będzie w jednym pliku?

Reply:
Wydaje mi się to bez sensu bo CharacterHealth jest częścią modelu, więc warto je dać do jednego pliku.

Reply:
Tak jak w przyszłości może być 60 zmiennych, które będą częścią modelu i wtedy mamy skrypt 1000 linijek

Reply:
Wydaje mi się to nie być szczególnym problemem, że będziemy mieć 1000 linijek kodu. Można to rozbić na 60 plików, nie wiem co byłoby bardziej czytelne.

Reply:
W sumie sam nie wiem co będzie lepsze, kieruję się tylko zasadą "Jedna klasa jeden plik".

Reply:
W takim razie kończę konwersację.